### PR TITLE
feat(wallet): Send Solana in UI

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Implemented Sending Solana in UI

1. Allows sending `Solana` in the UI
2. Allows viewing `Solana` transactions in the UI

note:(Adding and Sending `SPL` Tokens will be handled in this issue https://github.com/brave/brave-browser/issues/22039)

Resolves <https://github.com/brave/brave-browser/issues/21891>
Resolves <https://github.com/brave/brave-browser/issues/21892>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/161145648-7a9cf1a6-abe7-47a0-bfba-e803bd3d1fe6.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx [hunk 1]
```diff
@@ -118,12 +120,21 @@ function ConfirmTransactionPanel (props: Props) {
...
     updateUnapprovedTransactionSpendAllowance,
     updateUnapprovedTransactionNonce
   } = props
+  const {
+    solFeeEstimates,
+    defaultNetworks
+  } = useSelector(({ wallet }: { wallet: WalletState }) => wallet)
+  const dispatch = useDispatch()
+
+  const transactionsNetwork = React.useMemo(() => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx:134]: Extract this to its own hook. Logic is reused multiple times

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx:134]: Updated!

File: components/brave_wallet_ui/common/async/lib.ts [hunk 2]
```diff
@@ -227,20 +228,24 @@ export function refreshBalances () {
...
+            chainId: network.chainId
+          }
+          return network.chainId === BraveWallet.LOCALHOST_CHAIN_ID &&
+            getSolBalanceInfo.error !== 0
+            ? { ...emptyBalance, chainId: network.chainId }
+            : solBalanceInfo
+        }
+
         // MULTICHAIN: Backend needs to allow chainId for getSolanaBalance method
         // Will be implemented here https://github.com/brave/brave-browser/issues/21695
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/lib.ts]: Removed Comments!

File: components/brave_wallet_ui/common/hooks/transaction-parser.ts [hunk 3]
```diff
@@ -66,7 +67,7 @@ export interface ParsedTransaction extends ParsedTransactionFees {
   isApprovalUnlimited?: boolean
 }
 
-export function useTransactionFeesParser (selectedNetwork: BraveWallet.NetworkInfo, networkSpotPrice: string) {
+export function useTransactionFeesParser (selectedNetwork: BraveWallet.NetworkInfo, networkSpotPrice: string, solFeeEstimates?: SolFeeEstimates) {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/transaction-parser.ts:70]: ++ lets handle in this issue https://github.com/brave/brave-browser/issues/22041

**Review by josheleonard — APPROVED** [review]
LGTM

**Pull request merged**